File: kotlintest/src/main/java/com/android/test/base/ListFragmentAdapter.java
package com.android.test.base;

import android.view.ViewGroup;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class ListFragmentAdapter extends FragmentPagerAdapter {

    private List<BaseFragment> mFragmentList;
    private List<String> mTabNameList;
    private Fragment mCurFragment;

    public ListFragmentAdapter(FragmentManager fm, ArrayList<BaseFragment> linkFragmentLists, ArrayList<String> mTabNameList) {
        super(fm);
        this.mTabNameList = mTabNameList;
        this.mFragmentList = linkFragmentLists;
    }


    @Override
    public Fragment getItem(int position) {
        return mFragmentList.get(position);
    }

    @Override
    public int getCount() {
        return mFragmentList != null ? mFragmentList.size() : 0;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return mTabNameList.get(position);
    }

    @Override
    public void setPrimaryItem(ViewGroup container, int position, Object object) {
        super.setPrimaryItem(container, position, object);
        mCurFragment = (Fragment) object;
    }

    public Fragment getCurFragment() {
        return mCurFragment;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
//        super.destroyItem(container, position, object);
    }
}

File: app/src/main/java/com/example/app/life/ListFragmentAdapter.kt
package com.example.app.life

import android.view.View
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

/**
 * Create by liuxue on 2021/1/25 0025.
 * description:
 */
class ListFragmentAdapter(
    fm: FragmentManager,
    val fragmentLists: MutableList<Fragment>,
    val tabNameList: MutableList<String>
) :
    FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return fragmentLists.get(position)
    }

    override fun getCount(): Int {
        return fragmentLists?.let {
            it.size
        }
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return tabNameList?.let {
            it.get(position)
        }
    }

    override fun getItemPosition(`object`: Any): Int {
        var view: View = `object` as View
        return view.getTag() as Int
    }

}
